src/extractor/guidance/coordinate_extractor.cpp

```diff
@@ -736,21 +816,18 @@ CoordinateExtractor::TrimCoordinatesToLength(std::vector<util::Coordinate> coord
 
     if (use_cache && length_cache.back() < desired_length && coordinates.size() >= 2)
     {
```

Comment:
here, we had problems regarding the length cache. The cache was not in sync with the coordinates, resulting in potential problems when requiring both to be of the same size.